### ISSUE TITLE
JavaScript show-hide

### ISSUE THREAD

**Issue by zme1** [zme1/toscana#32]
I tried writing JavaScript to toggle the display of the tables related to the heat map, with about 90% success. I have one bug to work out, and I don't know how to approach it... The display toggle works perfectly when I load the page and click table cells, but I don't know how to hide the table display *again* once I click another cell. They accumulate on the bottom of the screen. I've attached my JavaScript below:

```
function showHide() {
    var tbody = document.getElementById('tbody-' + this.id);
    if (tbody.style.display == "none") {
        tbody.style.display = "table-row-group"
    } else {
        tbody.style.display = "none"
    }
}
```
@djbpitt This may look somewhat familiar to you; I used your election stacked-bar graph for inspiration, at @ebeshero 's request. 

I manually pasted and pushed all the related visualization code into my keyMembers.html page, so I can develop it locally.

[My full JavaScript file](https://github.com/zme1/toscana/blob/master/js/svg.js)

**Comment by zme1** [comment]
A sample `rect` element:

```
<rect class="heatRec" id="andreiniv191901" stroke="black" stroke-width=".2"
               x="9.00900900900901" y="-7.804878048780488" width="9.00900900900901"
               height="7.804878048780488" fill="rgb(255, 255, 212)"/>
```

And a sample corresponding `tbody`:

```
   <tbody class="heatToggle"
          id="tbody-andreiniv191901"
          style="display: none;">
      <tr>
         <td>#andreiniv</td>
         <td>1919-01</td>
         <td/>
      </tr>
   </tbody>
```

**Comment by ebeshero** [comment]
@zme1 Think about the sequence of events you want to happen on click: You want to fire a function that hides any showing table elements before you open the next one...

**Comment by zme1** [comment]
@ebeshero I wrote a JavaScript function that resets the style display back to none before firing anything else in the script above, but that then ostensibly locks all the style displays to none. Do I need to write something more complex than a simple "set the `tbody` display to none," or do I need to tell the browser to do something else?

**Comment by zme1** [comment]
I noticed that the election XML file had a function that seemed as if it were for this exact purpose, but I can't debug JavaScript well enough (for now, at least) to understand exactly why it's breaking.

**Comment by ebeshero** [comment]
@zme1 Yes--you're on the right track in reading the election file's JavaScript...and I think you're close to figuring it out (really!) so at this point, I'll make like the rubber duckie and say, "quack". 
'

**Comment by zme1** [comment]
@ebeshero Patched!!! It's amazing the power a single dot can have over a person sometimes.....

**Comment by ebeshero** [comment]
@zme1 Agreed! A single dot can make all the difference. "...one small drop of ink produces that which makes thousands, perhaps millions, think." --Lord Byron. (If only Lord Byron could have learned to code--but wait, his daughter kinda helped to set us all in motion, didn't she?) :-)

**Comment by ebeshero** [comment]
The Byron quote is even better when I look it up--it's in _Don Juan_:
```
But words are things, and a small drop of ink,      
Falling like dew, upon a thought, produces 
That which makes thousands, perhaps millions, think;      
’T is strange, the shortest letter which man uses 
Instead of speech, may form a lasting link      
Of ages; to what straits old Time reduces 
Frail man, when paper — even a rag like this, 
Survives himself, his tomb, and all that’s his.
```

**Comment by zme1** [comment]
@ebeshero The next item on my list is figuring out how to create the reading view of the texts..

**Comment by ebeshero** [comment]
@zme1 That ought to be the easy part with XSLT (particularly after all the data viz you've been working on...)

**Comment by zme1** [comment]
@ebeshero I thought as much, too. Today at my project meeting, the head translator said he would like to give the user the option to toggle any 2 of the 3 versions of the minutes we have (manuscripts, transcription, and translation), and I'm inclined to agree with him. That sounds like *more* JavaScript development to me!! It's just a matter of determining how that will work in oXygen, then deciding how to develop it spatially on the webpage.

**Comment by ebeshero** [comment]
HTML: 
* How do you want to divide up the minutes? A whole page for a whole year? A whole page for a month? 
* How are the manuscript images arranged? Do you imagine wanting to show multiple ms pages lined up next to text transcriptions? (I don't know if we talked about this, but it's a good use of the `@facs` attribute on the TEI `<pb/>` element: If you've encoded page-beginnings, you may want an `@facs="imageFile.jpg"` sitting on those, and you can use that association to help you position the code following that `<pb/>` element in relation to an associated image. 

CSS: Flexboxes to control positioning on the page
JavaScript: show/hide on click

Ideas: http://shelleygodwinarchive.org/sc/oxford/ms_abinger/c56/#/p40
http://dickinson16.newtfire.org/16/1607.html

**Comment by zme1** [comment]
@ebeshero I think it would be best to have one page for each year, with 7 subsidiary pages inside the Minutes page on the website.

I also haven't finished placing all the scans in my repository, but every meeting that I processed has its own folder in my `img` directory, and every scan has a filename of `pagenumber.png`. I think `jpg` is more compact and would probably help with load time, so should I proceed with `jpg` file types?

For the actual rendering, I think a one-to-one relationship between the documents would be easiest. That is, the pages or scans would show one page at a time, and if the user wanted to advance to the next page or change one entirely, that would be automatically processed for both of the versions on the screen. 

I think that, instead of directing the user to different pages for each meeting, I could encode radio buttons and/or check boxes to decide which meeting they want to view, and which 2 (or even 1) of the 3 versions of the minutes they would like to read. 

So, in a nutshell, 7 different sub-pages that contain some form of JavaScript enabled selection of individual meetings and mediums to explore.

**Comment by ebeshero** [comment]
@zme1 These are all good ideas. But don't bother changing your image format! Jpegs are pretty common photo image formats, but pngs are just as good if not better...(Did you know that png stands for "portable network graphic"?) I'd leave the format alone, but pay attention to how they appear in a browser window.
